module/applications/api/primary-sheet-mixin.mjs

```diff
@@ -0,0 +1,361 @@
+     * @typedef {object} SheetTabDescriptor5e
+     * @property {string} tab                       The tab key.
+     * @property {string} label                     The tab label's localization key.
+     * @property {string} [icon]                    A font-awesome icon.
+     * @property {string} [svg]                     An SVG icon.
+     * @property {SheetTabCondition5e} [condition]  A predicate to check before rendering the tab.
+     */
+
+    /**
+     * @callback SheetTabCondition5e
+     * @param {Document} doc  The Document instance.
+     * @returns {boolean}     Whether to render the tab.
+     */
+
+    /**
+     * Sheet tabs.
+     * @type {SheetTabDescriptor5e[]}
+     */
+    static TABS = [];
+
+    /**
+     * Available sheet modes.
+     * @enum {number}
+     */
+    static MODES = {
+      PLAY: 1,
+      EDIT: 2
+    };
+
+    /* -------------------------------------------- */
+
+    /**
+     * Dynamically configured part descriptors, backported from V13.
+     * @type {Readonly<Record<string, HandlebarsTemplatePart>>}
+     */
+    #partDescriptors;
+
+    /* -------------------------------------------- */
+    /*  Properties                                  */
+    /* -------------------------------------------- */
+
+    /**
+     * Filters for applied inventory sections.
+     * @type {Record<string, FilterState5e>}
+     */
+    _filters = {};
+
+    /* -------------------------------------------- */
+
+    /**
+     * The mode the sheet is currently in.
+     * @type {PrimarySheet5e.MODES|null}
+     * @protected
+     */
+    _mode = null;
+
+    /* -------------------------------------------- */
+    /*  Rendering                                   */
+    /* -------------------------------------------- */
+
+    /** @inheritDoc */
+    _configureRenderOptions(options) {
+      super._configureRenderOptions(options);
+
+      // Set initial mode
+      let { mode, renderContext } = options;
+      if ( (mode === undefined) && (renderContext === "createItem") ) mode = this.constructor.MODES.EDIT;
+      this._mode = mode ?? this._mode ?? this.constructor.MODES.PLAY;
+    }
+
+    /* -------------------------------------------- */
+
+    /** @inheritDoc */
+    _configureRenderParts(options) {
+      const parts = super._configureRenderParts(options);
+      for ( const key of Object.keys(parts) ) {
+        const tab = this.constructor.TABS.find(t => t.tab === key);
+        if ( tab?.condition && !tab.condition(this.document) ) delete parts[key];
+      }
+      return parts;
+    }
+
+    /* -------------------------------------------- */
+
+    /** @inheritDoc */
+    async _renderFrame(options) {
+      const html = await super._renderFrame(options);
+      const header = html.querySelector(".window-header");
+
+      if ( !game.user.isGM && this.document.limited ) {
+        html.classList.add("limited");
+        return html;
+      }
+
+      // Add edit <-> play slide toggle.
+      if ( this.isEditable ) {
```

Comment:
Consider moving this out to `_renderModeToggle` like we do with `_renderSource`. That would allow us to add/remove the toggle if the ownership of the Document changes.